var/spack/repos/builtin/packages/lci/package.py

```diff
@@ -0,0 +1,186 @@
+        default="auto",
+        values=is_positive_int,
+        description="Max number of completion queue entries that can be posted "
+        "(completion queue length) at the fabric layer by default",
+    )
+
+    variant("debug", default=False, description="Enable the debug mode")
+    variant("pcounter", default=False, description="Enable the performance counters")
+    variant(
+        "papi", default=False, description="Enable the PAPI plugin to collect hardware counters"
+    )
+
+    variant(
+        "enable-pm",
+        description="Process management backends to enable",
+        values=disjoint_sets(("auto",), ("pmix", "pmi2", "pmi1", "mpi", "local"))
+        .prohibit_empty_set()
+        .with_default("auto")
+        .with_non_feature_values("auto"),
+    )
+    variant(
+        "default-pm",
+        description="Order of process management backends to try by default",
+        values=disjoint_sets(("auto",), ("pmix", "pmi2", "pmi1", "mpi", "local"), ("cray",))
+        .prohibit_empty_set()
+        .with_default("auto")
+        .with_non_feature_values("auto"),
+    )
+
+    generator("ninja", "make", default="ninja")
+
+    depends_on("cmake@3.12:", type="build")
+    depends_on("libfabric", when="fabric=ofi")
+    depends_on("rdma-core", when="fabric=ibv")
+    depends_on("ucx", when="fabric=ucx")
+    depends_on("mpi", when="enable-pm=mpi")
+    depends_on("papi", when="+papi")
+    depends_on("doxygen", when="+docs")
+    depends_on("cray-pmi", when="default-pm=cray")
+
+    def cmake_args(self):
+        args = [
+            self.define_from_variant("LCI_SERVER", "fabric"),
+            self.define("LCI_FORCE_SERVER", True),
+            self.define_from_variant("LCI_WITH_EXAMPLES", "examples"),
+            self.define_from_variant("LCI_WITH_TESTS", "tests"),
+            self.define_from_variant("LCI_WITH_BENCHMARKS", "benchmarks"),
+            self.define_from_variant("LCI_WITH_DOC", "docs"),
+            self.define_from_variant("LCI_ENABLE_MULTITHREAD_PROGRESS", "multithread-progress"),
+            self.define_from_variant("LCI_DEBUG", "debug"),
+            self.define_from_variant("LCI_USE_PERFORMANCE_COUNTER", "pcounter"),
+            self.define_from_variant("LCI_USE_PAPI", "papi"),
+        ]
+
+        if not self.spec.satisfies("dreg=auto"):
+            arg = self.define_from_variant("LCI_USE_DREG_DEFAULT", "dreg")
+            args.append(arg)
+
+        if not self.spec.satisfies("enable-pm=auto"):
+            arg = self.define("LCT_PMI_BACKEND_ENABLE_PMI1", "enable-pm=pmi1" in self.spec)
+            args.append(arg)
+            arg = self.define("LCT_PMI_BACKEND_ENABLE_PMI2", "enable-pm=pmi2" in self.spec)
+            args.append(arg)
+            arg = self.define("LCT_PMI_BACKEND_ENABLE_MPI", "enable-pm=mpi" in self.spec)
+            args.append(arg)
+            arg = self.define("LCT_PMI_BACKEND_ENABLE_PMIX", "enable-pm=pmix" in self.spec)
+            args.append(arg)
```

Comment:
Can you just use `args.extend` here and elsewhere + a literal list as an argument?
```python
args.extend([
    self.define("LCT_PMI_BACKEND_ENABLE_PMI1", self.spec.satisfies("enable-pm=pmi1")),
    ...
])
```